### PR TITLE
Don’t crash if a compilation database is malformed JSON

### LINKED ISSUE #1019: Malformed `compile_commands.json` causes crash
[`tryLoadCompilationDatabase`](https://github.com/apple/sourcekit-lsp/blob/e8e4fb210616e8d41f5803f19dd6427fc27e247e/Sources/SKCore/CompilationDatabase.swift#L71-L93C2) fails aggressively if it finds a `compile_commands.json` but fails to parse it:

```
SKCore/CompilationDatabase.swift:84: Fatal error: 'try!' expression unexpectedly raised an error: Swift.DecodingError.dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid JSON.", underlyingError: Optional(Foundation.JSONError.unexpectedEndOfFile)))
```

Simple repro:
```
touch A.swift
touch compile_commands.json
code .
```
The language server crashes on startup.

### PR DESCRIPTION
**Pull request by ahoppen**
Fixes #1019
rdar://120804823

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by bnbarham — APPROVED** [review]

File: Sources/SKCore/CompilationDatabase.swift [hunk 1]
```diff
@@ -81,11 +81,11 @@ public func tryLoadCompilationDatabase(
       try! RelativePath(validating: "build"),
     ]
   return
-    try! searchPaths
+    searchPaths
```
[the remark this answers is not in the document]
Comment (ahoppen) [reply Sources/SKCore/CompilationDatabase.swift:84]: Good idea. Changed `try?` to `orLog`.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**